bondcpp/src/bond.cpp

```diff
@@ -437,21 +432,27 @@ bool Bond::waitUntilBroken(rclcpp::Duration timeout)
       break;  //  The deadline has expired
     }
     r.sleep();
+
+    std::unique_lock<std::mutex> lock(state_machine_mutex_);
+    if (sm_.getState().getId() == SM::Dead.getId()) {
+      broken = true;
+      break;
```

Comment:
I don't think this `break` is necessary, as we'll get out at the top of the loop.

Reply:
Done